### PR TITLE
Adds surgery examine messages

### LINKED ISSUE #19255: Surgery has terrible user feedback
Surgery, without a surgical computer, is absolutely awful because it gives you almost no feedback whatsoever.

You have no way to know what surgery is currently being performed. It does not show up with examining, checking yourself, sprites, or anything other than the surgical computer.

You have no way of knowing what the next surgical step is, and given that we both have a shitload of different kinds of surgery, each with different steps and amount of steps, it's very easy to get lost and difficult to pick up because of the above problem.

Knowing what works as a ghetto tool is impossible outside of either looking at the extremely outdated and sometimes downright wrong wiki, looking at the code, or just blindly testing for items that sound like they might work (but you usually end up chopping somebody up instead). There really should be a little tidbit in the descriptions of these items that state "Performs well as a ghetto bone saw for surgery" or "Performs poorly as a ghetto scalpel for surgery".

### PR DESCRIPTION
**Pull request by swindly**
:cl: Swindly
add: Added examine messages indicating that there are active surgeries for mobs.
/:cl:

[why]: It prevents people from smacking patients with surgical drapes for 5 minutes before realizing that there is already a surgery in progress.

Maybe closes https://github.com/tgstation/tgstation/issues/19255
Maybe there should be a different message for robotic limbs. I don't know.

### CODE REVIEW

**Comment by Repukan** [comment]
Why would you add this when there's a perfectly good screen computer telling you what to do

**Comment by swindly** [comment]
Because the computer isn't always around to tell people that they are split open, which is a reasonable thing for a person without medical equipment to determine.

**Comment by swindly** [comment]
>how to continue the operation

I only added 20 lines. Come on.

**Comment by Repukan** [comment]
>Year of our Lord and savior by his eternal grace, Jesus Christ of Nazareth, 2018

>Thinking I'll read the code

**Comment by swindly** [comment]
Nothing in the one line changelog indicated that the examine message would display the next step.

**Comment by subject217** [comment]
Repukan, maybe you should try reading the PR description before you write out an angry reply.

I've always thought that it was weird that you had no way to tell that a person was surgically cut open unless the computer said so. Good change.

**Comment by zombiegums** [comment]
Very good QoL change. It's kind of nuts how many people will get up from surgery without being closed up first.

File: code/modules/mob/mob_helpers.dm [hunk 1]
```diff
@@ -508,4 +508,18 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
...
+		return
+	for(var/datum/surgery/S in surgeries)
+		var/obj/item/bodypart/BP = S.operated_bodypart
+		var/zone = S.location
+		if(!get_location_accessible(src, zone))
+			continue
+		if(!BP)
+			. += "<span class='warning'>[src] has surgical markings where [p_their()] [parse_zone(zone)] should go!</span><br>"
+		else
+			. += "<span class='warning'>There are surgical markings on [p_their()] [BP.name]!</span><br>"
```
Comment (ShizCalev) [inline code/modules/mob/mob_helpers.dm]: put a new line at the end please

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by AnturK** [comment]
We really need some unification of examines between types, by sections or something similar.

File: code/modules/mob/mob_helpers.dm [hunk 2]
```diff
@@ -508,4 +508,18 @@ It's fairly easy to fix if dealing with single letters but not so much with comp
...
+	. = ""
+	if(!surgeries.len)
+		return
+	for(var/datum/surgery/S in surgeries)
+		var/obj/item/bodypart/BP = S.operated_bodypart
+		var/zone = S.location
+		if(!get_location_accessible(src, zone))
+			continue
+		if(!BP)
+			. += "<span class='warning'>[src] has surgical markings where [p_their()] [parse_zone(zone)] should go!</span><br>"
```
Comment (AnturK) [inline code/modules/mob/mob_helpers.dm]: Shouldn't this src be He/She/It helper ? Most examine messages only state the name once.

File: code/modules/mob/living/simple_animal/slime/slime.dm [hunk 3]
```diff
@@ -409,6 +409,7 @@
 
 	var/msg = "<span class='info'>*---------*\nThis is [icon2html(src, user)] \a <EM>[src]</EM>!\n"
 	if (src.stat == DEAD)
+		msg += surgeries_examine()
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/slime/slime.dm]: This should probably show for live animals too.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/slime/slime.dm (hunk 3 above)
Comment (swindly) [reply code/modules/mob/living/simple_animal/slime/slime.dm]: Only ded slimes can get surgery but okay

**Comment by AnturK** [comment]
?

**Pull request closed without merging**